### PR TITLE
chore(dashboard): text changes and fixes

### PR DESCRIPTION
**Pull request by shahednasser**
- Improvements to existing texts
- Add subtitles to sections where useful.
- Added the ability to edit the message shown in a list when no records are found + change the message where necessary
- Change documentation link to `https://docs.medusajs.com/v2`.

### CODE REVIEW

File: packages/admin-next/dashboard/src/i18n/translations/en.json [hunk 1]
```diff
@@ -1111,12 +1148,12 @@
       "details": "Promotion Details"
     },
     "tabs": {
-      "template": "Template",
-      "promotion": "Promotion",
+      "template": "Type",
+      "promotion": "Details",
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/i18n/translations/en.json]: I think we should update they keys as well

File: packages/admin-next/dashboard/src/i18n/translations/en.json [hunk 2]
```diff
@@ -1111,12 +1148,12 @@
...
       "campaign": "Campaign"
     },
     "fields": {
-      "method": "Method",
+      "method": "Campaign",
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/i18n/translations/en.json]: Update key as well

File: packages/admin-next/dashboard/src/i18n/translations/en.json [hunk 3]
```diff
@@ -1130,11 +1167,11 @@
       "conditions": {
         "rules": {
           "title": "Who can use this code?",
-          "description": "Is the customer allowed to add the promotion code? Discount code can be used by all customers if left untouched."
+          "description": "Which customer is allowed to use the promotion code? Promotion code can be used by all customers if left untouched."
         },
         "target-rules": {
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/i18n/translations/en.json:1171]: not related to your PR, but keys shouldn't use kebab case. But feel free to leave it, and we can clean it up later.

**Review by kasperkristensen — APPROVED** [review]
LGTM just some minor nits

File: packages/admin-next/dashboard/src/i18n/translations/en.json (hunk 2 above)
Comment (shahednasser) [reply packages/admin-next/dashboard/src/i18n/translations/en.json]: Found an exisiting `campaign` key so removed this one instead

**Pull request merged**